### PR TITLE
Add Editors Unit tests from VSO to github

### PR DESCRIPTION
**Pull request by basoundr**
This change moves over the Editors Unit Tests from VSO to Github

This change:

- Adds couple of unit tests projects to the solution
- Makes both the projects SDK based
- Excludes all the tests from running. Currently the references(disabled in this PR) are not PackageReference based. The references are commented out. Later change needs to happen to convert them over to Package References.

I will open another issue when checking this change in to track the work to fix the references and enable the tests again.

@dotnet/project-system @davkean @srivatsn for review

### CODE REVIEW

File: src/Microsoft.VisualStudio.Editor.UnitTests/AuthoringTests.txt [hunk 1]
```diff
@@ -0,0 +1,131 @@
+﻿==========================================================================
+    Visual Studio Team System: Overview of Authoring and Running Tests
```
Comment (RaulPerez1) [inline src/Microsoft.VisualStudio.Editor.UnitTests/AuthoringTests.txt]: >Visual Studio Team System: Overview of Authoring and Running Tests [](start = 4, length = 66)

Skimming through the text, this looks outdated. I'd just remove it.

File: src/Microsoft.VisualStudio.Editor.UnitTests/Microsoft.VisualStudio.Editors.UnitTests.csproj [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+    <UsingProjectVSCommonVslangproj2>true</UsingProjectVSCommonVslangproj2>
+    <UsingProjectVSCommonVslangproj80>true</UsingProjectVSCommonVslangproj80>
+    <UsingPartitionVsdata>true</UsingPartitionVsdata>
+    <UsingProjectVsdataMicrosoftVisualstudioData>true</UsingProjectVsdataMicrosoftVisualstudioData>
+    <UsingProjectVsdataMicrosoftVisualstudioDataInterop>true</UsingProjectVsdataMicrosoftVisualstudioDataInterop>
+    <UsingProjectVsdataMicrosoftVisualStudioQualityToolsMockObjectFramework>true</UsingProjectVsdataMicrosoftVisualStudioQualityToolsMockObjectFramework>
+    <UsingPartitionVsip>true</UsingPartitionVsip>
+    <UsingProjectVsipMicrosoftVisualstudioShellDesign>true</UsingProjectVsipMicrosoftVisualstudioShellDesign>
+    <UsingPartitionXmake>true</UsingPartitionXmake>
+    <UsingProjectXmakeXmaketasks>true</UsingProjectXmakeXmaketasks>
```
Comment (RaulPerez1) [inline src/Microsoft.VisualStudio.Editor.UnitTests/Microsoft.VisualStudio.Editors.UnitTests.csproj]: All this is razzle stuff that you should wipe before it goes into github. I think some of the stuff below is as well

**Comment by RaulPerez1** [comment]
> Project("{778DAE3C-4631-46EA-AA77-85C1314464D9}") = "SettingsDesignerUnitTests", "Microsoft.VisualStudio.Editors.SettingsDesigner.UnitTests\SettingsDesignerUnitTests.vbproj", "{A03E115A-9193-455D-A70A-CFEB943D0319}"

This should be the full name for consistency.

---
Refers to: src/ProjectSystem.sln:62 in 770279d. [](commit_id = 770279daee7cb728b13c2c083085e2b0cd9db2ad, deletion_comment = False)

File: src/Microsoft.VisualStudio.Editor.UnitTests/ApplicationDesigner/ApplicationDesignerPanelTests.cs [hunk 3]
```diff
@@ -0,0 +1,545 @@
+using System;
```
Comment (RaulPerez1) [inline src/Microsoft.VisualStudio.Editor.UnitTests/ApplicationDesigner/ApplicationDesignerPanelTests.cs]: This goes for all the source files - Do you need to copyright notice before they go into GitHub?

File: src/Microsoft.VisualStudio.Editor.UnitTests/AuthoringTests.txt (hunk 1 above)
Comment (basoundr) [inline src/Microsoft.VisualStudio.Editor.UnitTests/AuthoringTests.txt]: With this change, I am bringing in all the files. This is just a port of the projects itself. There will be a separate pass(which will take some time) to clean all of these files and remove unwanted Razzle variables. 

This change will build a test binary containing nothing in it.

File: src/Microsoft.VisualStudio.Editor.UnitTests/Microsoft.VisualStudio.Editors.UnitTests.csproj (hunk 2 above)
Comment (basoundr) [inline src/Microsoft.VisualStudio.Editor.UnitTests/Microsoft.VisualStudio.Editors.UnitTests.csproj]: Reason above

File: src/Microsoft.VisualStudio.Editor.UnitTests/ApplicationDesigner/ApplicationDesignerPanelTests.cs (hunk 3 above)
Comment (basoundr) [inline src/Microsoft.VisualStudio.Editor.UnitTests/ApplicationDesigner/ApplicationDesignerPanelTests.cs]: I will need to do this.

**Comment by basoundr** [comment]
Naming the `SettingsDesignerUnitTests` per convention will also be part of the clean up

File: src/Microsoft.VisualStudio.Editor.UnitTests/ApplicationDesigner/ApplicationDesignerPanelTests.cs (hunk 3 above)
Comment (basoundr) [inline src/Microsoft.VisualStudio.Editor.UnitTests/ApplicationDesigner/ApplicationDesignerPanelTests.cs]: Done

**Comment by basoundr** [comment]
Got verbal 👍 from @davkean

**Pull request merged**